cmd/swagger/commands/diff/spec_analyser_test.go

```diff
@@ -110,21 +104,53 @@ func makeTestCases(t testing.TB, matches []string) []testCaseData {
 				})
 		}
 	}
-	return testCases
-}
 
-func linesInFile(t testing.TB, fileName string) io.ReadCloser {
-	file, err := os.Open(fileName)
-	require.NoError(t, err)
-	return file
+	return testCases
 }
 
 func TestIssue2962(t *testing.T) {
 	oldSpec := filepath.Join("..", "..", "..", "..", "fixtures", "bugs", "2962", "old.json")
 	newSpec := filepath.Join("..", "..", "..", "..", "fixtures", "bugs", "2962", "new.json")
-	diffs, err := getDiffs(oldSpec, newSpec)
+
+	t.Run("should diff", func(t *testing.T) {
+		diffs, err := getDiffs(oldSpec, newSpec)
+		require.NoError(t, err)
+
+		const (
+			expectedChanges  = 3
+			expectedBreaking = 1
+		)
+
+		t.Run(fmt.Sprintf("should find %d breaking changes", expectedBreaking), func(t *testing.T) {
+			require.Len(t, diffs, expectedChanges)
+			require.Equal(t, expectedBreaking, diffs.BreakingChangeCount())
+		})
+	})
+}
+
+func fixturePath(file string, parts ...string) string {
+	return filepath.Join("..", "..", "..", "..", "fixtures", "diff", strings.Join(append([]string{file}, parts...), ""))
+}
+
+type testCaseData struct {
+	name          string
+	oldSpec       string
+	newSpec       string
+	expectedLines io.Reader
+	expectedFile  string
+}
+
+func fixturePart(file string) string {
+	base := filepath.Base(file)
+	parts := strings.Split(base, ".diff.txt")
+	return parts[0]
+}
+
+func linesInFile(t testing.TB, fileName string) io.ReadCloser {
+	t.Helper()
+
+	file, err := os.Open(fileName)
 	require.NoError(t, err)
 
-	require.Len(t, diffs, 3)
-	require.Equal(t, 1, diffs.BreakingChangeCount())
+	return file
```

Comment:
(only looking at the diff without the full context) does this need a `file.Close()` somewhere (could be in a `t.Cleanup()` ?